src/PowerShell/AzPSScriptBuilder.ts

```diff
@@ -0,0 +1,97 @@
+
+export default class AzPSScriptBuilder {
+
+    static getLatestModulePathScript(moduleName: string): string {
+        let script = `try {
+            $ErrorActionPreference = "Stop"
+            $WarningPreference = "SilentlyContinue"
+            $output = @{}
+            $latestModulePath = (Get-Module -Name '${moduleName}' -ListAvailable | Sort-Object Version -Descending | Select-Object -First 1).ModuleBase
+            $latestModulePath = Join-Path $latestModulePath ".." | Resolve-Path
+            $output['${AzPSConstants.Result}'] = $latestModulePath.ToString()
+            $output['${AzPSConstants.Success}'] = "true"
+        }
+        catch {
+            $output['${AzPSConstants.Error}'] = $_.exception.Message
+        }
+        return ConvertTo-Json $output`;
+
+        return script;
+    }
+
+    static async getAzPSLoginScript(loginConfig: LoginConfig) {
+        let loginMethodName = "";
+        let commands = 'Clear-AzContext -Scope Process; ';
+        commands += 'Clear-AzContext -Scope CurrentUser -Force -ErrorAction SilentlyContinue; ';
+
+        if (loginConfig.environment.toLowerCase() == "azurestack") {
+            commands += `Add-AzEnvironment -Name '${loginConfig.environment}' -ARMEndpoint '${loginConfig.resourceManagerEndpointUrl}' | out-null;`;
+        }
+        if (loginConfig.authType === "service_principal") {
+            if (loginConfig.servicePrincipalKey) {
+                let servicePrincipalKey: string = loginConfig.servicePrincipalKey.split("'").join("''");
+                commands += AzPSScriptBuilder.loginWithSecret(loginConfig.environment, loginConfig.tenantId, loginConfig.servicePrincipalId, servicePrincipalKey);
+                loginMethodName = 'service principal with secret';
+            } else {
+                await loginConfig.getFederatedToken();
+                commands += AzPSScriptBuilder.loginWithOIDC(loginConfig.environment, loginConfig.tenantId, loginConfig.servicePrincipalId, loginConfig.federatedToken);
+                loginMethodName = "OIDC";
+            }
+        } else {
+            if (loginConfig.servicePrincipalId) {
+                commands += AzPSScriptBuilder.loginWithUserAssignedIdentity(loginConfig.environment, loginConfig.servicePrincipalId);
+                loginMethodName = 'user-assigned managed identity';
+            } else {
+                commands += AzPSScriptBuilder.loginWithSystemAssignedIdentity(loginConfig.environment);
+                loginMethodName = 'system-assigned managed identity';
+            }
+        }
+
+        if (!loginConfig.allowNoSubscriptionsLogin && loginConfig.subscriptionId) {
+            if (loginConfig.tenantId) {
+                commands += `Set-AzContext -SubscriptionId '${loginConfig.subscriptionId}' -TenantId '${loginConfig.tenantId}' | out-null;`;
+            } else {
+                commands += `Set-AzContext -SubscriptionId '${loginConfig.subscriptionId}' | out-null;`;
+            }
+        }
```

Comment:
Shall we extract this to a function called `setSubscription` and add a warning message when the arguments `allow-no-subscriptions` and `subscription-id` are both missed?

Reply:
- Extract to a new function: done.
- Add a warning message: I think it should be added in **LoginConfig**.